### ISSUE TITLE
Theory: nearly completely unusable fullscreen mode

### ISSUE THREAD

**Issue by soocki** [QupZilla/qupzilla#2048]
Fullscreen mode is not very usable, in my opinion it needs a lot of improvement. There is couple things wrong with, bear with me while I make my point.

Firstly, when the top menu/bar is accessed, say by moving the mouse cursor into it's position it causes the whole webpage to shift down. When the cursor is moved back down onto the webpage it shift back up. This behaviour is wrong for a couple of reasons:
- It makes it nearly impossible to hit weblinks accurately when moving the cursor down since before the cursor gets to the link the page is shifted up. There is a slight delay in the shift of the page and the user need to wait until it happens before moving the cursor to the desired weblink position.
- The whole page position shift just doesn't look right, the menu should just overlap the webpage instead, it would look 'right' (I realise this statement might sound subjective but I feel that most users would agree with it), as well as fix the first issue described in the previous paragraph.

Secondly the menu is not accesible through the GUI when in fullscreen mode, the 'menu' button changes into 'fullscreen toggle' button so in order to access the menu, the fullscreen mode needs to be disabled first. The whole point of enabling the fullscreen mode is to be able to work comfortably in it and since every time the menu is to be accessed it needs to be toggled on and off it makes me wonder if it's worth enabling in the first place.

This completes my theory of completely (or perhaps almost completely) unusable fullscreen mode.

**Comment by innermous** [comment]
I agree with thesis about menu button. Although, I don't think overlapping is a good idea, you won't be able reach the topmost links because menu will simply overlap it. That's why firefox has same model for fullscreen mode.
Second thought is totally right, as I already said. I sign under it, definitely needs changes.

**Comment by soocki** [comment]
If we were to compare Qupzilla to Other Browsers than Chromium have an overlapping menu. There, I said it, balance restored!

The way I see it Qupzilla is special a special browser and it should do things it's own way and behave in a way it's users want it to behave. I realize it will be hard to please everyone, perhaps impossible to do so. You can please some of the people all of the time, you can please all of the people some of the time but you definitely cannot please all of the people all of the time. Anyway, back to the point.

I do find overlapping menu much more convinient after trying both. Is it just a matter of preference? Perhaps so, yet somehow I think most people would agree with overlapping menu being more convinient to use. Saying that I do realize I only see this from my own point of view so it's only an assumption based on my own opinions, which are extremely subjective in nature.

The llinks located at the top of the webpage can be accesed after a slight delay when the menu slides back into position, this way at least you don't need to wait for all of the links, all of the time, just the one's on top of the webpage and only when you need to access those links. The way fullscreen mode behaves now I find it extremely difficult to use and most of the time I end up switching back to windowed mode. I have a whole openbox panel dedicated only for Qupzilla, so I definitely could use it in fullscreen mode at all times. As things stand I simply cannot do that, missing on a lot of pixels here man and keep in mind that in my case there's only 1366 by 768 so 1.049*10^6 of them!

**Comment by innermous** [comment]
>Other Browsers than Chromium have an overlapping menu

FF doesn't.

**Comment by 99epep** [comment]
Hi, I'm new to QupZilla (since few hours), and thinking adopting it for my notebook, but my screen is only 600px high, fullscreen is necessary.

For me, major drawback is the lack of menu access in full screen mode. I'll give a try to other lightweight browser waiting this issue to be resolved.


I also had difficulties catching some links on top of pages, and think it can be worked out with simple modifications :

- The navigation bar pops down when the pointer is near top of the screen (about 5 pixels from top), why not waiting it really hits the top ?

- The navigation bar hides when the pointer is out after a 1s timeout, wouldn't it be more natural to hide it when the cursor goes far enough ?


There is a third point disturbing me, when opening a new tab, we can immediately type URL, blindly ! Is it possible to show the navigation bar in new tab until a page is loaded ?

Thanks !

**Comment by soocki** [comment]
I'm totally witch you on that one mate, the fullscreen mode needs a lot of improvement, sad thing is not many people seems to care/notice this. Glad we on the same side anyhow.

**Comment by 99epep** [comment]
I did not saw version 2 was out when writing the previous message, can someone say us if fullscreen mode is better on this last version ? (I'm on 1.8.9)

I can't afford building it for my Debian, too much struggle with Qt …

**Comment by soocki** [comment]
I am on 2.0.1-2, as far as I can tell not much have changed.

**Comment by 99epep** [comment]
Thank you soocki !

So all I can do now is praying for us to be heard, and thank the dev team for all the work allready done.
